cpg_workflows/stages/realign_genotype_with_dragen.py

```diff
@@ -0,0 +1,329 @@
+        sequencing_group: SequencingGroup,
+    ) -> cpg_utils.Path:
+        sg_bucket: cpg_utils.Path = sequencing_group.dataset.prefix()
+        return (
+            sg_bucket
+            / GCP_FOLDER_FOR_RUNNING_PIPELINE
+            / f'{sequencing_group.name}_pipeline_cancelled_at{slugify(str(datetime.now()))}.json'
+        )
+
+    def queue_jobs(self, sequencing_group: SequencingGroup, inputs: StageInput) -> StageOutput | None:
+        cancel_pipeline_run: PythonJob = get_batch().new_python_job(
+            name='CancelIcaPipelineRun',
+            attributes=(self.get_job_attrs() or {}) | {'tool': 'Dragen'},
+        )
+        cancel_pipeline_run.image(image=image_path('cpg_workflows'))
+        outputs = self.expected_outputs(sequencing_group=sequencing_group)
+        cancel_pipeline = cancel_pipeline_run.call(
+            cancel_ica_pipeline_run.run,
+            ica_pipeline_id_path=str(inputs.as_path(target=sequencing_group, stage=AlignGenotypeWithDragen)),
+            api_root=ICA_REST_ENDPOINT,
+        ).as_json()
+        get_batch().write_output(
+            cancel_pipeline,
+            str(outputs),
+        )
+        return self.make_outputs(
+            target=sequencing_group,
+            data=outputs,
+            jobs=cancel_pipeline_run,
+        )
+
+
+@stage(
+    analysis_type='ica_data_download',
+    required_stages=[PrepareIcaForDragenAnalysis, MonitorAlignGenotypeWithDragen, GvcfMlrWithDragen],
+)
+class DownloadDataFromIca(SequencingGroupStage):
+    def expected_outputs(
+        self,
+        sequencing_group: SequencingGroup,
+    ) -> cpg_utils.Path:
+        bucket_name: cpg_utils.Path = sequencing_group.dataset.prefix()
+        return bucket_name / GCP_FOLDER_FOR_ICA_DOWNLOAD / f'{sequencing_group.name}'
+
+    def queue_jobs(self, sequencing_group: SequencingGroup, inputs: StageInput) -> StageOutput | None:
+        bucket_name: str = get_path_components_from_gcp_path(path=str(object=sequencing_group.cram))['bucket']
+
+        batch_instance: Batch = get_batch()
+        ica_download_job: BashJob = batch_instance.new_bash_job(
+            name='DownloadDataFromIca',
+            attributes=(self.get_job_attrs() or {}) | {'tool': 'ICA'},
+        )
+        ica_analysis_folder_id_path: str = batch_instance.read_input(
+            inputs.as_path(target=sequencing_group, stage=PrepareIcaForDragenAnalysis),
+        )
+        ica_download_job.storage(storage=calculate_needed_storage(cram=str(sequencing_group.cram)))
+        ica_download_job.memory('8Gi')
+        ica_download_job.image(image=image_path('ica'))
+
+        # Download an entire folder with ICA. Don't log projectId or API key
+        authenticate_cloud_credentials_in_job(ica_download_job)
+        ica_download_job.command(
+            f"""
+            {ICA_CLI_SETUP}
+            icav2 projectdata download $(cat {ica_analysis_folder_id_path} | jq -r .analysis_output_fid) {sequencing_group.name} --exclude-source-path
```

Comment:
Aha, this the corresponding reading in of that JSON output. Seems fine, not sure if `$(cat analysis_id)` would be less work, but you might need to worry about line endings. 

Basically I've never gotten used to JQ, it scares me.